core/dataURI/dataURI.go

```diff
@@ -79,15 +85,22 @@ func StoreData(uri string, dataReader io.Reader, dataLength uint32) (int64, comm
 	if trace.IsLogging(logger.TRACE) {
 		trace.Trace("Storing data at %s", uri)
 	}
+
 	dataURI, err := url.Parse(uri)
 	if err != nil || !strings.EqualFold(dataURI.Scheme, "file") {
 		return 0, &Error{"Invalid data URI"}
 	}
 
-	filePath := dataURI.Path + ".tmp"
-	file, err := os.OpenFile(filePath, os.O_WRONLY|os.O_CREATE, 0600)
+	baseFilePath, err := filepath.Abs(filepath.Clean(dataURI.Path))
+	if err != nil {
+		return 0, &Error{"Failed to resolve file path"}
```

Comment:
Can you change the returned error to something like `&Error{fmt.Sprintf("Failed to resolve file path %v", dataURI.Path)}` here and all the other places?